api/src/main/java/io/minio/MinioClient.java

```diff
@@ -3939,6 +3940,129 @@ public void setBucketPolicy(String bucketName, String policy)
     response.body().close();
   }
 
+  /**
+   * Set XML string of LifeCycle on a given bucket.
+   * Delete the lifecycle of bucket in case a null is passed as lifeCycle.
+   * @param bucketName   Bucket name.
+   * @param lifeCycle    Bucket policy XML string.
+   *
+   * </p><b>Example:</b><br>
+   * <pre>{@code String lifecycle  = "<LifecycleConfiguration><Rule><ID>expire-bucket</ID><Prefix></Prefix>"
+   * + "<Status>Enabled</Status><Expiration><Days>365</Days></Expiration></Rule></LifecycleConfiguration>";
+   *
+   * setBucketLifecycle("my-bucketname", lifecycle); }</pre>
+   * @throws InvalidBucketNameException  upon invalid bucket name is given
+   * @throws NoSuchAlgorithmException    upon requested algorithm was not found during
+   *                                     signature calculation
+   * @throws InsufficientDataException   upon getting EOFException while reading given
+   *                                     InputStream even before reading given length
+   * @throws IOException                 upon connection error
+   * @throws InvalidKeyException         upon an invalid access key or secret key
+   * @throws NoResponseException         upon no response from server
+   * @throws XmlPullParserException      upon parsing response xml
+   * @throws ErrorResponseException      upon unsuccessful execution
+   * @throws InternalException           upon internal library error
+   */
+  public void setBucketLifeCycle(String bucketName, String lifeCycle)
+          throws InvalidBucketNameException, NoSuchAlgorithmException,
+          InsufficientDataException, IOException, InvalidKeyException, NoResponseException,
+          XmlPullParserException, ErrorResponseException, InternalException {
+    // If lifecycle is empty or null then delete it
+    if (lifeCycle == null || "".equals(lifeCycle)) {
+      deleteBucketLifeCycle(bucketName);
```

Comment:
You would need to error out or S3 service to error out, not to delete with `Set()`

Reply:
done.